AWS-CodePipeline/two-environment-pipeline-template/{{cookiecutter.outputDir}}/codepipeline.yaml

```diff
@@ -0,0 +1,814 @@
+
+  CodeBuildProjectBuildAndPackage:
+    Condition: IsMainBranchPipeline
+    Type: AWS::CodeBuild::Project
+    Properties:
+      Artifacts:
+        Type: CODEPIPELINE
+      Environment:
+        Type: LINUX_CONTAINER
+        ComputeType: BUILD_GENERAL1_SMALL
+        Image: aws/codebuild/amazonlinux2-x86_64-standard:3.0
+        PrivilegedMode: true
+        EnvironmentVariables:
+          - Name: SAM_TEMPLATE
+            Value: !Ref SamTemplate
+          - Name: TESTING_REGION
+            Value: !Ref TestingRegion
+          - Name: PROD_REGION
+            Value: !Ref ProdRegion
+          - Name: TESTING_PIPELINE_EXECUTION_ROLE
+            Value: !Ref TestingPipelineExecutionRole
+          - Name: PROD_PIPELINE_EXECUTION_ROLE
+            Value: !Ref ProdPipelineExecutionRole
+          - Name: TESTING_ARTIFACT_BUCKET
+            Value: !Ref TestingArtifactBucket
+          - Name: PROD_ARTIFACT_BUCKET
+            Value: !Ref ProdArtifactBucket
+          - Name: TESTING_IMAGE_REPOSITORY
+            Value: !Ref TestingImageRepository
+          - Name: PROD_IMAGE_REPOSITORY
+            Value: !Ref ProdImageRepository
+      ServiceRole: !GetAtt CodeBuildServiceRole.Arn
+      Source:
+        Type: CODEPIPELINE
+        BuildSpec:  |
+          version: 0.2
+          phases:
+            install:
+              runtime-versions:
+                python: 3.8
+              commands:
+                - pip install --upgrade pip
+                - pip install --upgrade awscli aws-sam-cli
+                - nohup /usr/local/bin/dockerd --host=unix:///var/run/docker.sock --host=tcp://127.0.0.1:2375 --storage-driver=overlay2 &
+                - timeout 15 sh -c "until docker info; do echo .; sleep 1; done"
+            build:
+              commands:
+                - sam build --use-container --template ${SAM_TEMPLATE}
+                - . ./assume-role.sh ${TESTING_PIPELINE_EXECUTION_ROLE} test-package
+                - sam package --s3-bucket ${TESTING_ARTIFACT_BUCKET}
+                              {%- if cookiecutter.testing_image_repository %}
+                              --image-repository ${TESTING_IMAGE_REPOSITORY}
+                              {%- endif %}
+                              --region ${TESTING_REGION}
+                              --output-template-file packaged-test.yaml
+                - . ./assume-role.sh ${PROD_PIPELINE_EXECUTION_ROLE} prod-package
+                - sam package --s3-bucket ${PROD_ARTIFACT_BUCKET}
+                              {%- if cookiecutter.testing_image_repository %}
+                              --image-repository ${PROD_IMAGE_REPOSITORY}
+                              {%- endif %}
+                              --region ${PROD_REGION}
+                              --output-template-file packaged-prod.yaml
+          artifacts:
+            files:
+              - packaged-test.yaml
+              - packaged-prod.yaml
+              - assume-role.sh
```

Comment:
I'm not sure why assume-role.sh is in the artifacts

Reply:
because the deploy stage requires all these 3 files. The source is not accessible there due to the artifact is not source.

```
InputArtifacts:
                  - Name: BuildArtifactAsZip
```